vissl/data/data_helper.py

```diff
@@ -24,6 +24,65 @@ def get_mean_image(crop_size):
     return img
 
 
+@contextlib.contextmanager
+def with_temporary_numpy_seed(seed: int):
+    """
+    Context manager to run a specific portion of code with a given seed:
+    resumes the numpy state after the execution of the block
+    """
+    random_state = np.random.get_state()
+    np.random.seed(seed)
+    yield
+    np.random.set_state(random_state)
+
+
+def unbalanced_sub_sampling(
+    total_size: int, nb_samples: int, skip_samples: int = 0, seed: int = 0
+) -> np.ndarray:
+    """
+    Given an original dataset of size 'total_size', sub_sample part of the dataset such that
+    the sub sampling is deterministic (identical across distributed workers)
+    Return the selected indices
+    """
+    with with_temporary_numpy_seed(seed):
+        return np.random.choice(
+            total_size, size=skip_samples + nb_samples, replace=False
+        )[skip_samples:]
+
+
+def balanced_sub_sampling(
+    labels: np.ndarray, nb_samples: int, skip_samples: int = 0, seed: int = 0
+) -> np.ndarray:
+    """
+    Given some labels, sub_sample a part of the labels such that:
+    - the number of samples of each label differs by at most one
+    - the sub sampling is deterministic (identical across distributed workers)
+    Return the selected indices
+    """
+    groups = {}
+    for i, label in enumerate(labels):
+        groups.setdefault(label, []).append(i)
+
+    unique_labels = sorted(groups.keys())
+    sq, sr = divmod(skip_samples, len(unique_labels))
+    q, r = divmod(nb_samples, len(unique_labels))
```

Comment:
is it feasible to make the `sq, sr, q, r` more expanded names? :)

Reply:
Indeed... sorry for that. I wanted to do it, but it slipped my mind and forgot to do it. I will fix this !